frontend/src/components/SelectApplication.vue

```diff
@@ -26,45 +37,49 @@ onMounted(async () => {
         return Promise.reject(error);
     }
 });
+
+function filterEnv(selectedEnv: string) {
+    environmentFilter.value = selectedEnv;
+    selectedApplication.value = null;
+}
 </script>
 
 <template>
     <PageTitle :displaySelectedApplication="false"></PageTitle>
 
     <form id="selectApplicationForm" class="form-container">
         <div v-if="applicationsUserAdministers.length">
-            <!-- Disabled - not needed for MVP -->
-            <!-- Logic in script section:
-        const environments = [
-          "Prod", "Test", "Dev" // Dynamically assemble from list of apps.
-        ]
-
-        const environmentFilter = ref<string>("Prod")
-        -->
-            <!--
-        <div class="row">
-          <label>Filter by environment</label>
-        </div>
-        <div class="row">
-          <div class="col-auto" v-for="env in environments">
-            <input type="radio" id="r-env-{{env}}" name="envRadio" class="form-check-input" :value="env"
-              v-model="environmentFilter" :checked="(environmentFilter === env)" />
-              &nbsp;
-            <label class="form-check-label" for="r-env-{{env}}">{{env}}</label>
-          </div>
-        </div>
-        -->
+            <div class="row">
+                <label>Filter by environment</label>
+            </div>
+            <RadioGroup
+                :options="environments"
+                :initialValue="environmentFilter"
+                @change="filterEnv"
+            ></RadioGroup>
             <div class="row">
                 <div class="form-group col-md-5">
                     <label>Select the application to administer</label>
-
                     <select
                         id="applicationSelect"
                         class="form-select"
                         v-model="selectedApplication"
                         :size="applicationsUserAdministers.length + 1"
                     >
                         <option
+                            v-if="environmentFilter !== ALL_APPS"
+                            v-for="app in applicationsUserAdministers.filter(
+                                (item) =>
+                                    item.app_environment?.includes(
+                                        environmentFilter.toUpperCase()
+                                    )
+                            )"
+                            :value="app"
+                        >
+                            {{ app.application_description }}
+                        </option>
```

Comment:
Hi Nick,
I feel putting this javascript filtering code in the <script></script> area with a function might be cleaner, not at the html <template>.
Also you probably don't need "v-if" and "v-else" condition. It can be done in one function filtering. Just move your `environmentFilter == ALL_APPS` inside filter to return 'true' for all element no matter what it is.

Reply:
I have changed this as suggested.